src/libcrun/linux.c

```diff
@@ -1947,6 +1947,81 @@ do_mounts (libcrun_container_t *container, int rootfsfd, const char *rootfs, con
   return 0;
 }
 
+/*
+ * libcrun_container_do_bind_mount
+ *
+ *  Allows external plugins and handlers to perform bind `mounts` on container.
+ *  returns: 0 if successful anything else states `error` and configures `err` with relevent error.
+ */
+int
+libcrun_container_do_bind_mount (libcrun_container_t *container, int rootfsfd, const char *rootfs, char *mount_source, char *mount_destination, char **mount_options, size_t mount_options_len, libcrun_error_t *err)
+{
+  int ret;
+  size_t rootfs_len = get_private_data (container)->rootfs_len;
+
+  const char *target = consume_slashes (mount_destination);
+  cleanup_free char *data = NULL;
+  unsigned long flags = 0;
+  unsigned long extra_flags = 0;
+  cleanup_close int targetfd = -1;
+  int is_dir = 1;
+  uint64_t rec_clear = 0;
+  uint64_t rec_set = 0;
+
+  get_private_data (container)->rootfs = rootfs;
+  get_private_data (container)->rootfsfd = rootfsfd;
+  get_private_data (container)->rootfs_len = rootfs ? strlen (rootfs) : 0;
```

Comment:
we are overriding the previous values, that we will leak

Reply:
At this point the old fd is already in a bad state hence updating otherwise `mount` later fails with `crun: invalid rootfs state`. But I think we already discussed this part hence marking as resolved. Please mark this as unresolved if this is not resolved.